### PR TITLE
[ci] Move to arcade , remove GitInfo and Mono dependencies

### LINKED ISSUE #19185: [build] Remove GitInfo
### Description

Sometimes we see some failures on ci related with the usage of GitInfo.
We just use it with a small subset, can we write code to get the exact same info? 

We could also look at switching to
https://github.com/dotnet/Nerdbank.GitVersioning

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

dotnet/maui

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

I was *not* able test on other platforms

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

```shell
C:\Users\cloudtest\.nuget\packages\gitinfo\2.2.0\build\GitInfo.targets(342,3): error MSB3491: Could not write lines to file "obj\Release\net8.0-android\GitIsDirty.cache". The process cannot access the file 'D:\a\_work\1\s\src\Controls\src\Core\obj\Release\net8.0-android\GitIsDirty.cache' because it is being used by another process. [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net8.0-android]
```

### LINKED ISSUE #20814: [ci] Remove mono as requirement
### Description

Right now we have some requirements on our CI that need mono. I tried to [remove here](https://github.com/dotnet/maui/pull/20576) , and then I saw some failures later on fresh machines and I had to add it [back here.](https://github.com/dotnet/maui/pull/20813). 

We should remove and rewrite all our pipeline  to try to use `dotnet test` and `dotnet nuget` . I have identified the following places not having mono fails

- Device tests  for android for example need mono to start the `NUnit.ConsoleRunner`
- Pack on macOS needs mono to start the `NuGet.CommandLine.6.6.1`

### Relevant log output

```shell
Time Elapsed 00:00:53.65
Run UITests lib /Users/builder/azdo/_work/2/s/src/Compatibility/ControlGallery/test/Android.UITests/bin/Debug/net472/Microsoft.Maui.Controls.Android.UITests.dll
Add Where filter to NUnit cat==Issues
Could not resolve path for tool "mono" using these directories: /Users/builder/Library/Developer/Xamarin/android-sdk-macosx/emulator,/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/platform-tools,/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/cmdline-tools/latest/bin,/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/cmdline-tools/7.0/bin,/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/cmdline-tools/5.0/bin,/Users/builder/Library/Developer/Xamarin/android-sdk-macosx/tools/bin,/usr/local/microsoft/powershell/7,/Users/builder/azdo/_work/2/s/bin/dotnet,/Users/builder/.dotnet/tools,/Users/builder/azdo/_work/_tool/dotnet,/usr/local/bin,/usr/local/share/dotnet,~/.dotnet/tools,/Library/Frameworks/Mono.framework/Versions/Current/Commands,/usr/bin,/bin,/usr/sbin,/sbin
/Users/builder/azdo/_work/2/s/eng/devices/tools/NUnit.ConsoleRunner.3.16.3/tools/nunit3-console.exe is a .NET Framework executable, you might need to install Mono for it to execute successfully.
Executing: /Users/builder/azdo/_work/2/s/eng/devices/tools/NUnit.ConsoleRunner.3.16.3/tools/nunit3-console.exe "/Users/builder/azdo/_work/2/s/src/Compatibility/ControlGallery/test/Android.UITests/bin/Debug/net472/Microsoft.Maui.Controls.Android.UITests.dll" --where "cat==Issues" "--work=/Users/builder/azdo/_work/2/a/test-results/android" "--out=/Users/builder/azdo/_work/2/a/test-results/android/run_uitests_output-133531728934878300.log" "--config=Debug"
An error occurred when executing task 'cg-uitest'.
Completed in 00:02:45.3767402
```

```shell
dotnet-pack-additional
========================================
Executing task: dotnet-pack-additional
Could not resolve path for tool "mono" using these directories: /Users/builder/azdo/_work/2/s/bin/dotnet,/usr/local/microsoft/powershell/7,/Users/builder/.dotnet/tools,/Users/builder/azdo/_work/_tool/dotnet,/usr/local/bin,/usr/local/share/dotnet,~/.dotnet/tools,/Library/Frameworks/Mono.framework/Versions/Current/Commands,/usr/bin,/bin,/usr/sbin,/sbin
/Users/builder/azdo/_work/2/s/tools/NuGet.CommandLine.6.6.1/tools/NuGet.exe is a .NET Framework executable, you might need to install Mono for it to execute successfully.
Executing: /Users/builder/azdo/_work/2/s/tools/NuGet.CommandLine.6.6.1/tools/NuGet.exe install "_NativeAssets.windows" -OutputDirectory "/Users/builder/azdo/_work/2/s/artifacts/additional-assets" -Version "0.0.0-commit.e2c5c86249621857107c779af0f79b4d06613766.655" -ExcludeVersion -Source "https://aka.ms/skiasharp-eap/index.json" -NonInteractive
An error occurred when executing task 'dotnet-pack-additional'.
Completed in 00:00:00.1964573
```

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

This enables or CI to use [arcade](https://github.com/dotnet/arcade). Setups a new way to do the versioning relying on arcade rules. Right now this doesn't enables signing/shipping with arcade, but only building using part of the arcade infrastructure.

Fixes #19185
Fixes #20814 

Some variables from [arcade props](https://github.com/dotnet/arcade/blob/8ec8057ac5073b6b2e3fcb0a33d588d2a3357ad3/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L47) want to start using where we can on msbuild:

Here are the key changes:

Changes to Build Properties and Targets:

* [`Directory.Build.props`](diffhunk://#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eL2-R38): Replaced the import of `eng\Versions.props` with `Sdk.props` and added new property groups. Some properties were removed in later parts of the file. [[1]](diffhunk://#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eL2-R38) [[2]](diffhunk://#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eL130-L131) [[3]](diffhunk://#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eL144-R183)
* [`Directory.Build.targets`](diffhunk://#diff-50e91c82311ea26f2a73202525dfdf2b0a89c178ee666b2bf3ad4c84ac4c06dcL3-R16): Removed some imports and added others, including `Sdk.targets`. Some package information was also removed. [[1]](diffhunk://#diff-50e91c82311ea26f2a73202525dfdf2b0a89c178ee666b2bf3ad4c84ac4c06dcL3-R16) [[2]](diffhunk://#diff-50e91c82311ea26f2a73202525dfdf2b0a89c178ee666b2bf3ad4c84ac4c06dcL84) [[3]](diffhunk://#diff-50e91c82311ea26f2a73202525dfdf2b0a89c178ee666b2bf3ad4c84ac4c06dcL97-L99)
* [`eng/Environment.Build.props`](diffhunk://#diff-5bcc4573a88e5f9f54ddabe6070f983daaa0024a78517340159156d0837e35f4L12-L33): Removed several property groups, including those related to testing particular set of renderers and language version.
* `eng/Git.Build.targets` and `eng/Nuget.targets`: Entire files were removed. [[1]](diffhunk://#diff-e09d512e4fa963c3e8f6d6b7cbb0a88145a2bdd47e02f7eace15795f3211bea3L1-L6) [[2]](diffhunk://#diff-82ad065b0b533e8bdc2ac7ebd5b2c39735b49a710f02b55c530b8754d19d5babL1-L16)
* [`eng/Publishing.props`](diffhunk://#diff-350198cd1e3783d062a0d2dcee549e02f83dae123340fa4be238637ee421b527R1-R5): New file added with a property group indicating that the project produces .NET release shipping assets.

Changes to Solution Files:

* `Microsoft.Maui-dev.sln`, `Microsoft.Maui-mac.slnf`, and `Microsoft.Maui.sln`: Added new testing projects and made corresponding changes in the global sections. [[1]](diffhunk://#diff-56a1a0f9c52a56a926536aeaea1bbc7d999ab8e6249f7583dfe29e1678bb8241R246-R249) [[2]](diffhunk://#diff-56a1a0f9c52a56a926536aeaea1bbc7d999ab8e6249f7583dfe29e1678bb8241R623-R626) [[3]](diffhunk://#diff-56a1a0f9c52a56a926536aeaea1bbc7d999ab8e6249f7583dfe29e1678bb8241R739-R740) [[4]](diffhunk://#diff-922389d363b81b1d09a9873b148c5ab83ec77eac3b60c2fd725e96d483b293e0R30) [[5]](diffhunk://#diff-922389d363b81b1d09a9873b148c5ab83ec77eac3b60c2fd725e96d483b293e0R46) [[6]](diffhunk://#diff-e40652dbfb03e597de53ae81d2e9131698e74489ff1876678d2481ff2355003cR264-R269) [[7]](diffhunk://#diff-e40652dbfb03e597de53ae81d2e9131698e74489ff1876678d2481ff2355003cR671-R678) [[8]](diffhunk://#diff-e40652dbfb03e597de53ae81d2e9131698e74489ff1876678d2481ff2355003cR800-R802)

Other Changes:

* [`GitInfo.txt`](diffhunk://#diff-9cba54c1b529e0e82538de01335cbe4638bdf59abe15343f2bde8bbccbb10b58L1): Removed the version information.
* [`NuGet.config`](diffhunk://#diff-9c5952957709545aad811b400e7e516eebade1e44fc4fbc23203403ffeb92c34R6): Added `dotnet-tools` as a new package source.
* `build.cake` and `build.ps1`: Made modifications related to the restoration of Cake tool and building Cake arguments. [[1]](diffhunk://#diff-c6eace5a70f0c2d920dfa97e988d4cad3d0131dd2b029bb2646c96222e866cf8L30) [[2]](diffhunk://#diff-3258bc4162690adead72c70b672093e68c92e86a4628148f0015d61738f21b5aL12-R16)

### CODE REVIEW

File: Microsoft.Maui-mac.slnf [hunk 1]
```diff
@@ -42,6 +42,7 @@
       "src\\Graphics\\src\\Graphics\\Graphics.csproj",
       "src\\Graphics\\src\\Text.Markdig\\Graphics.Text.Markdig.csproj",
       "src\\Graphics\\tests\\DeviceTests\\Graphics.DeviceTests.csproj",
+      "src\\Graphics\\tests\\Graphics.Tests\\Graphics.Tests.csproj",
```
Comment (rmarinho) [inline Microsoft.Maui-mac.slnf:45]: this is needed because build needs this to be built to run tests against the dll after

File: Microsoft.Maui-dev.sln [hunk 2]
```diff
@@ -245,6 +245,10 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Controls.SourceGen.UnitTest
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SourceGen.UnitTests", "src\Controls\tests\SourceGen.UnitTests\SourceGen.UnitTests.csproj", "{BC7F7C82-694F-4B97-86FC-273FB3FACA25}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{054C5BAC-3671-4F76-B32E-47CF4E64BB39}"
```
Comment (rmarinho) [inline Microsoft.Maui-dev.sln:248]: so the build on the Mac builds the Compatibility.Core.UnitTests also

File: eng/cake/helpers.cake [hunk 3]
```diff
@@ -120,39 +120,4 @@ bool IsTarget(string target) =>
     Argument<string>("target", "Default").Equals(target, StringComparison.InvariantCultureIgnoreCase);
 
 bool TargetStartsWith(string target) =>
-    Argument<string>("target", "Default").StartsWith(target, StringComparison.InvariantCultureIgnoreCase);
-
-void RunTestsNunit(string unitTestLibrary, NUnit3Settings settings)
```
Comment (rmarinho) [inline eng/cake/helpers.cake:125]: We now use dotnet test

File: eng/devices/android.cake [hunk 4]
```diff
@@ -387,37 +413,18 @@ Task("cg-uitest")
...
+            PROJECT.FullPath,
+            config: CONFIGURATION,
+            pathDotnet: DOTNET_PATH,
+            noBuild: false,
+            resultsFileNameWithoutExtension: resultName,
+            filter: Argument("filter", ""));
 
 	// When all tests are inconclusive the run does not fail, check if this is the case and fail the pipeline so we get notified
-	FailRunOnOnlyInconclusiveTests(System.IO.Path.Combine(nunitSettings.Work.FullPath, "TestResult.xml"));
+	//FailRunOnOnlyInconclusiveTests(System.IO.Path.Combine(nunitSettings.Work.FullPath, "resultName.xml"));
```
Comment (rmarinho) [inline eng/devices/android.cake:427]: Not sure we need this anymore

File: src/Controls/tests/UITests/UITest.cs [hunk 5]
```diff
@@ -36,7 +36,9 @@ protected UITest(TestDevice testDevice) : base(testDevice)
 				ciArtifactsDirectory = Path.Combine(ciArtifactsDirectory, "Controls.AppiumTests");
 
 			string assemblyDirectory = Path.GetDirectoryName(System.AppDomain.CurrentDomain.BaseDirectory)!;
-			string projectRootDirectory = Path.GetFullPath(Path.Combine(assemblyDirectory, "..", "..", ".."));
+			string repoRootDirectory = Path.GetFullPath(Path.Combine(assemblyDirectory, "..", "..", "..", "..", ".."));
```
Comment (rmarinho) [inline src/Controls/tests/UITests/UITest.cs:39]: I wonder if we can use $(MauiRootDirectory) or something

File: src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs [hunk 6]
```diff
@@ -71,28 +74,41 @@ public void ProductAndCompany(string assemblyName)
 			Assert.AreEqual(s_company, actual.CompanyName);
 		}
 
-		static string GetFileFromRoot(string file)
+		internal static string GetFilePathFromRoot(string file)
```
Comment (rmarinho) [inline src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs:77]: @mattleibow @jonathanpeppers  you have any ideas to improve these tests where we need to reference the root directory , seems this is really easy to break.

Comment (jonathanpeppers) [reply src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs:77]: It's not very pretty, but you could make an MSBuild target write `$(MSBuildThisFileDirectory)` in a `.cs` file.

Something like:
```xml
<!-- untested, from memory -->
<WriteLinesToFile File="$(IntermediateOutputPath)TestConstants.cs" Lines="class TestConstants { public const string Directory = &quot;$(MSBuildThisFileDirectory)&quot;; }" WriteOnlyWhenDifferent="true" />
<ItemGroup>
  <Compile Include="$(IntermediateOutputPath)TestConstants.cs" />
  <FileWrites Include="$(IntermediateOutputPath)TestConstants.cs" />
</ItemGroup>
```

If running from the `.csproj`, this would point to its directory.

File: src/BlazorWebView/src/WindowsForms/Microsoft.AspNetCore.Components.WebView.WindowsForms.csproj [hunk 7]
```diff
@@ -16,7 +16,7 @@
     <!-- NuGet package information -->
     <IsPackable>true</IsPackable>
     <PackageId>Microsoft.AspNetCore.Components.WebView.WindowsForms</PackageId>
-    <PackageIconFile>$(MauiRootDirectory)Assets\aspnet-icon.png</PackageIconFile>
+    <PackageIcon>$(MauiRootDirectory)Assets\aspnet-icon.png</PackageIcon>
```
Comment (mattleibow) [inline src/BlazorWebView/src/WindowsForms/Microsoft.AspNetCore.Components.WebView.WindowsForms.csproj:19]: Why was this changed?

File: src/SingleProject/Resizetizer/test/UnitTests/Resizetizer.UnitTests.csproj [hunk 8]
```diff
@@ -8,13 +8,11 @@
...
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
     <PackageReference Include="Microsoft.Build.Framework" Version="16.7.0" PrivateAssets="all" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="16.7.0" PrivateAssets="all" />
     <PackageReference Include="SkiaSharp" />
     <PackageReference Include="SkiaSharp.Extended" />
-    <PackageReference Include="xunit" Version="2.5.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.1" />
+    <PackageReference Include="xunit" Version="$(XunitPackageVersion)"/>
     <PackageReference Include="coverlet.collector" Version="6.0.0" />
     <PackageReference Include="Svg.Skia" />
```
Comment (mattleibow) [inline src/SingleProject/Resizetizer/test/UnitTests/Resizetizer.UnitTests.csproj:17]: Is the `Microsoft.NET.Test.Sdk` package not needed? I see this is always added in the xunit templates.

ANd `xunit.runner.visualstudio`? Is that not needed for the IDE anymore?

Comment (rmarinho) [reply src/SingleProject/Resizetizer/test/UnitTests/Resizetizer.UnitTests.csproj:17]: Nop, those are added by the arcade sdk

File: src/BlazorWebView/src/WindowsForms/Microsoft.AspNetCore.Components.WebView.WindowsForms.csproj (hunk 7 above)
Comment (rmarinho) [reply src/BlazorWebView/src/WindowsForms/Microsoft.AspNetCore.Components.WebView.WindowsForms.csproj:19]: Yeah this is because the arcade sdk also sets that property, setting PackageIcon, it reads this Prop and uses that, but I will confirm the packages have the correct icon.

Comment (rmarinho) [reply src/BlazorWebView/src/WindowsForms/Microsoft.AspNetCore.Components.WebView.WindowsForms.csproj:19]: Good catch @mattleibow  seems something got wrong with the icon, do you have a way to compare 2 nupkg ? like one from main agains one from the PR see where they diff? 
[Microsoft.Maui.Controls.8.0.40-ci.net8.27947.zip](https://github.com/dotnet/maui/files/15083161/Microsoft.Maui.Controls.8.0.40-ci.net8.27947.zip)


Screenshot 2024-04-23 at 22 16 44

Comment (rmarinho) [reply src/BlazorWebView/src/WindowsForms/Microsoft.AspNetCore.Components.WebView.WindowsForms.csproj]: Screenshot 2024-04-24 at 12 52 49

Just confirmed with nupkg from this PR

File: global.json [hunk 9]
```diff
@@ -1,9 +1,10 @@
 {
   "tools": {
-    "dotnet": "7.0.203"
+    "dotnet": "9.0.100-preview.3.24204.13"
```
Comment (mattleibow) [inline global.json:3]: Is 5his correct? main is now net9?

File: src/Controls/tests/Xaml.UnitTests/MSBuild/_Directory.Build.props [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <MauiRootDirectory>$(MSBuildThisFileDirectory)..\..\..\..\..\..\..\..\..\</MauiRootDirectory>
+    <MauiRootDirectory>$(MSBuildThisFileDirectory)..\..\..\..\..\..\..\</MauiRootDirectory>
```
Comment (mattleibow) [inline src/Controls/tests/Xaml.UnitTests/MSBuild/_Directory.Build.props:4]: Is this changed because the output is no longer the bin in the folder but at the root?

File: src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs [hunk 11]
```diff
@@ -43,24 +43,27 @@ public void AssemblyVersion(string assemblyName)
...
-			Assert.AreEqual(expected.Minor, actual.FileMinorPart, $"FileMinorPart is wrong. {actual.ToString()}");
-			// Fails locally
-			//Assert.AreEqual(expected.Build, actual.FileBuildPart, $"FileBuildPart is wrong. {actual.ToString()}");
-			//We need to enable this
-			//	Assert.AreEqual(ThisAssembly.Git.Commits, version.FilePrivatePart);
-			Assert.AreEqual(s_productName, actual.ProductName);
-			Assert.AreEqual(s_company, actual.CompanyName);
-		}
+		// [Test, TestCaseSource(nameof(references))]
+		// public void FileVersion(string assemblyName)
```
Comment (mattleibow) [inline src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs:47]: Why was this disabled?

File: eng/Signing.props [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+  <ItemGroup Condition="'$(PrepareArtifacts)' == 'true'">
+    <ItemsToSignWithPaths Include="$(DownloadDirectory)**/*.msi" />
+    <ItemsToSignWithPaths Include="$(DownloadDirectory)**/*.exe" />
+    <ItemsToSignWithPaths Include="$(DownloadDirectory)**/*.nupkg" />
+    <ItemsToSignWithPaths Include="$(DownloadDirectory)**/*.zip" />
+
+    <ItemsToSignWithoutPaths Include="@(ItemsToSignWithPaths->'%(Filename)%(Extension)')" />
+    <ItemsToSignPostBuild Include="@(ItemsToSignWithoutPaths->Distinct())" Condition="'$(PostBuildSign)' == 'true'" />
+    <ItemsToSign Include="@(ItemsToSignWithPaths->Distinct())" Condition="'$(PostBuildSign)' != 'true'" />
+  </ItemGroup> -->
```
Comment (mattleibow) [inline eng/Signing.props:43]:
```suggestion
  </ItemGroup>
```

Comment (mattleibow) [inline eng/Signing.props:43]: Not sure if this is a valid comment...

image

Is there a reason why this block should be commented out?

File: src/Workload/Microsoft.NET.Sdk.Maui.Manifest/Microsoft.NET.Sdk.Maui.Manifest.csproj [hunk 13]
```diff
@@ -108,21 +108,21 @@
...
+    <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(Version)/$(PackageVersion).json" />
     <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(Version)/$(PackageReferenceVersion).json" />
-    <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(GitSha).json" />
-    <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(GitSemVerLabel).json" />
-    <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(GitBranch).json" />
+    <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(SourceRevisionId).json" />
+    <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(PackageVersion).json" />
+    <!-- <Copy SourceFiles="$(OutputPath)Rollback.json" DestinationFiles="$(PackageOutputPath)/metadata/rollbacks/$(GitBranch).json" /> -->
```
Comment (mattleibow) [inline src/Workload/Microsoft.NET.Sdk.Maui.Manifest/Microsoft.NET.Sdk.Maui.Manifest.csproj:117]: I think the branch one is the one the other teams use more than the versions. Is there no value for this in arcade?

File: eng/cake/dotnet.cake [hunk 14]
```diff
@@ -227,13 +227,17 @@ Task("dotnet-test")
             "**/Essentials.UnitTests.csproj",
             "**/Resizetizer.UnitTests.csproj",
             "**/Graphics.Tests.csproj",
-            "**/Compatibility.Core.UnitTests.csproj",
+            //"**/Compatibility.Core.UnitTests.csproj",
```
Comment (mattleibow) [inline eng/cake/dotnet.cake:230]: Is this intentional? I see you are using this project in the if block below.

File: eng/devices/android.cake [hunk 15]
```diff
@@ -234,13 +231,42 @@ Task("Test")
...
 		if (apps.Any()) {
 			TEST_APP = apps.FirstOrDefault().FullPath;
 		} else {
 			apps = GetFiles(binDir + "/*.apk");
+			var arcadeBin = new DirectoryPath("../../artifacts/bin/");
+
+			if(PROJECT.FullPath.Contains("Controls.DeviceTests"))
+			{
+				binDir = MakeAbsolute(new DirectoryPath(arcadeBin + "/Controls.DeviceTests/" + CONFIGURATION + "/" + TARGET_FRAMEWORK +  "/"));
+			}
```
Comment (mattleibow) [inline eng/devices/android.cake:246]: Can this maybe be made more generic and just use the project name without the extension?

File: eng/devices/android.cake [hunk 16]
```diff
@@ -472,19 +479,36 @@ void SetupAppPackageNameAndResult()
...
-			else {
-				Error("Error: Couldn't find .apk file");
-				throw new Exception("Error: Couldn't find .apk file");
+			if (apps.Count() == 0) 
+			{
+				var arcadeBin = new DirectoryPath("../../artifacts/bin/");
+				if(TEST_APP_PROJECT.FullPath.Contains("Controls.Sample.UITests"))
+				{
+					binDir = MakeAbsolute(new DirectoryPath(arcadeBin + "/Controls.Sample.UITests/" + CONFIGURATION + "/" + TARGET_FRAMEWORK));
+				}
```
Comment (mattleibow) [inline eng/devices/android.cake:494]: Same here, to make it more generic and future proof, we should use project name instead.

File: eng/devices/android.cake (hunk 15 above)
Comment (mattleibow) [reply eng/devices/android.cake:246]: Same for all the other platform cakes.

File: Directory.Build.props [hunk 17]
```diff
@@ -1,5 +1,41 @@
...
+    <SignAssembly>false</SignAssembly>
+    <MauiRootDirectory>$(MSBuildThisFileDirectory)</MauiRootDirectory>
+    <MauiSrcDirectory>$(MSBuildThisFileDirectory)src/</MauiSrcDirectory>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- NuGet package information -->
+    <Copyright>$(CopyrightNetFoundation)</Copyright>
+    <Company>Microsoft</Company>
+    <Product>Microsoft MAUI</Product>
```
Comment (mattleibow) [inline Directory.Build.props:19]:
```suggestion
    <Product>Microsoft .NET MAUI</Product>
```

Do we have an official name?

**Review by mattleibow — COMMENTED** [review]
Lots of work here and many changes, but it looks mostly OK. I have a few comments/questions on things.

File: global.json (hunk 9 above)
Comment (rmarinho) [reply global.json:3]: Yeah this is just tools, not dotnet for building maui, same as we use xharness 9.0.0 on main also

File: src/Controls/tests/Xaml.UnitTests/MSBuild/_Directory.Build.props (hunk 10 above)
Comment (rmarinho) [reply src/Controls/tests/Xaml.UnitTests/MSBuild/_Directory.Build.props:4]: yeah the bin directory is now somthing like artifacts/bin/Project/Debug for example , i think we need to create a new task to redo these tests that use like thes strong typed paths?

File: src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs (hunk 11 above)
Comment (rmarinho) [reply src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs:47]: By default the FileVersion is 42.42..42 on debug builds , so this testt needs to be rewrriten 
https://github.com/dotnet/arcade/blob/e543752a4d92d8df16a9f95b89fcfdc50aeabf74/Documentation/CorePackages/Versioning.md?plain=1#L238

File: eng/Signing.props (hunk 12 above)
Comment (rmarinho) [reply eng/Signing.props:43]: I didn't eneable signing  with arcade yet, that would be a next step

File: src/Workload/Microsoft.NET.Sdk.Maui.Manifest/Microsoft.NET.Sdk.Maui.Manifest.csproj (hunk 13 above)
Comment (rmarinho) [reply src/Workload/Microsoft.NET.Sdk.Maui.Manifest/Microsoft.NET.Sdk.Maui.Manifest.csproj:117]: If so, what teams? For what porpose? is this the performance team for example @jonathanpeppers  do you know if to run the dashboards then need the roolback ? 

Do we have this documented somewhere? 

I think the only rollback usage would be install the specific version for example when we release, but the branch ne might be needed, let's investigate. 

If so we can add the BUILD_BRANCHNAME from CI

File: eng/cake/dotnet.cake (hunk 14 above)
Comment (rmarinho) [reply eng/cake/dotnet.cake:230]: Yeah didn't figure why these fail to run for now, we can enable after it's not important. But i will take a look again

File: eng/devices/android.cake (hunk 15 above)
Comment (rmarinho) [reply eng/devices/android.cake:246]: yeah i still had in my last task if everything is working to clean up this. or figure  if it's a better way to find that path

File: src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs (hunk 11 above)
Comment (mattleibow) [reply src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs:47]: Should we rather wrap it in a `#if !DEBUG` since we don't really need this test in debug builds.

File: Directory.Build.props (hunk 17 above)
Comment (rmarinho) [reply Directory.Build.props]: Ok I took this, is correct right @davidortinau ? I also fixed [our unit test for this ](https://github.com/dotnet/maui/pull/21487/commits/bfec68b484858bb67a784b49e14fcae85c95c73b)

File: eng/cake/dotnet.cake (hunk 14 above)
Comment (rmarinho) [reply eng/cake/dotnet.cake]: Ok I figured this one, the unit test project was missing from the .sln that the build cake target runs before. We should look at a better way for this, I wonder how this worked before. 

https://github.com/dotnet/maui/pull/21487/commits/a5af6c058463c72e4a2c99709ab5398afee71e88

File: src/Workload/Microsoft.NET.Sdk.Maui.Manifest/Microsoft.NET.Sdk.Maui.Manifest.csproj (hunk 13 above)
Comment (rmarinho) [reply src/Workload/Microsoft.NET.Sdk.Maui.Manifest/Microsoft.NET.Sdk.Maui.Manifest.csproj]: Screenshot 2024-05-02 at 12 29 42
OK fixed this one .. this is merge on PRs but will be correct on a branch build.

**Review by mattleibow — COMMENTED** [review]
Looking close!

I just got the comment to not remove the test but just use `#if !DEBUG`:
 - https://github.com/dotnet/maui/pull/21487#discussion_r1577804533

And then the one to clean up the cake files to not have an if block for each project.
